### PR TITLE
MethodExtensions class-name is case-insensitive

### PR DESCRIPTION
**Pull request by staabm**
refs https://github.com/phpstan/phpstan/issues/9714#issuecomment-1660896676

### CODE REVIEW

File: tests/PHPStan/Rules/Exceptions/data/unthrown-exception.php [hunk 1]
```diff
@@ -544,3 +544,35 @@ public function doBar(string $s)
 	}
 
 }
+
+class TestCaseInsensitiveClassNames
```
Comment (staabm) [inline tests/PHPStan/Rules/Exceptions/data/unthrown-exception.php:548]: there was no exisiting bug to fix for the throw-type-extension case.. but adding test coverage can not hurt

tests DynamicMethodThrowTypeExtension

File: tests/PHPStan/Analyser/data/closure-return-type-extensions.php [hunk 2]
```diff
@@ -18,3 +18,6 @@
 
 $returnType = $closure->call($newThis, new class {});
 assertType('true', $returnType);
+
+$staticallyBoundClosureCaseInsensitive = \closure::bind($closure, $newThis);
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/closure-return-type-extensions.php:22]: test DynamicStaticMethodReturnTypeExtension

File: tests/PHPStan/Rules/Exceptions/data/missing-exception-method-throws.php [hunk 3]
```diff
@@ -95,4 +95,9 @@ public function dateTimeZoneDoesThrows(string $tz): void
 		new \DateTimeZone($tz);
 	}
 
+	public function dateTimeZoneDoesNotThrowCaseInsensitive(): void
+	{
+		new \DaTetImezOnE('UTC');
```
Comment (staabm) [inline tests/PHPStan/Rules/Exceptions/data/missing-exception-method-throws.php:100]: test DynamicStaticMethodThrowTypeExtension

File: tests/PHPStan/Analyser/data/bug-9714.php [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+use function PHPStan\Testing\assertType;
+
+class ExampleClass
+{
+	public function exampleFunction(): void
+	{
+		$xml = new \SimpleXmlElement('');
+		$elements = $xml->xpath('//data');
+		assertType('array<SimpleXmlElement>', $elements);
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/bug-9714.php:13]: test DynamicMethodReturnTypeExtension-case

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**